integrations/integrations_test.go

```diff
@@ -0,0 +1,190 @@
+package integrations
+
+import (
+	"context"
+	"io"
+	"net"
+	"strings"
+	"testing"
+	"time"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/store/snapshots"
+	storetypes "cosmossdk.io/store/types"
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	bftclient "github.com/cometbft/cometbft/rpc/client/http"
+	bfttypes "github.com/cometbft/cometbft/types"
+	dbm "github.com/cosmos/cosmos-db"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/server/api"
+	serverconfig "github.com/cosmos/cosmos-sdk/server/config"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/cosmos/gogoproto/grpc"
+	testapp "github.com/polymerdao/monomer/testapp"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+)
+
+// Application Constructor `appCreator` for testing
+func mockAppCreator(
+	_ log.Logger,
+	db dbm.DB,
+	_ io.Writer,
+	_ servertypes.AppOptions,
+) servertypes.Application {
+	app, err := testapp.New(db, "1")
+	if err != nil {
+		panic(err)
+	}
+
+	return &WrappedTestApp{App: app}
+}
+
+// Unit test for Monomer's custom `StartCommandHandler` callback
+func TestStartCommandHandler(t *testing.T) {
+	svrCtx := server.NewDefaultContext()
+	svrCtx.Config.RootDir = t.TempDir()
+	svrCtx.Config.DBBackend = "memdb"
+
+	// This flag must be set, because by default it's set to ""
+	svrCtx.Viper.Set("minimum-gas-prices", "0.025stake")
+	// This flag must be set to load the Monomer genesis file
+	viper.Set(monomerGenesisPathFlag, "./testdata/genesis.json")
+	// This flag must be set to configure Monomer's Engine Websocket
+	viper.Set(monomerEngineWSFlag, "127.0.0.1:8089")
+
+	clientCtx := client.Context{}
+	inProcessConsensus := true
+	opts := server.StartCmdOptions{
+		DBOpener: func(rootDir string, backendType dbm.BackendType) (dbm.DB, error) {
+			return dbm.NewMemDB(), nil
+		},
+	}
+
+	cmtListenAddr := svrCtx.Config.RPC.ListenAddress
+	cmtListenAddr = strings.TrimPrefix(cmtListenAddr, "tcp://")
+
+	go func() {
```

Comment:
Hmm. I tried to follow the same pattern we use `node/http.go` but the test seems to hang. Maybe I'm doing something wrong, but this is what I had:
```go
var wg conc.WaitGroup
defer wg.Wait()
wg.Go(func() {
    err := StartCommandHandler(svrCtx, clientCtx, mockAppCreator, inProcessConsensus, opts)
    require.NoError(t, err)
})
```
Do we need to use an error channel / `select` block?

Reply:
I think this is expected. Here's what I think is happening:

1. `StartCommandHandler` runs in the goroutine
2. It sets up all the services and then blocks on the signal channel
3. No signal is ever sent
4. We wait forever in the test for `StartCommandHandler` to finish

I think we will need an ugly hack to make this work. Instead of defining the signal channel in `StartCommandHandler`, we can make it a global variable.

```golang
var sigChan = make(chan signal.Signal, 1) // not sure if I got the naming right, but you get the idea
```

Now, we can send a signal on this channel at the end of the test to unblock `StartCommandHandler`. Does this make sense?

Reply:
Yes this makes sense, thank you. Here's what I have so far, but the test is still hanging ...

```go
// integrations.go

var sigCh = make(chan os.Signal, 1)

// ...

func StartCommandHandler(...) {
    // ...
    signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
    <-sigCh
}
```

```go
func TestStartCommandHandler(...) {
    // ...
    var wg conc.WaitGroup
    defer wg.Wait()
    wg.Go(func() {
        err := StartCommandHandler(svrCtx, clientCtx, mockAppCreator, inProcessConsensus, opts)
        require.NoError(t, err)
    })
    // ...
    sigCh <- syscall.SIGINT
}
```

Reply:
Where is it hanging?

Reply:
It logs that the Tx broadcasted successfully, so it appears to hang while waiting for a signal on the channel